core/modules/qdisp/LargeResultMgr.cc

```diff
@@ -0,0 +1,142 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2017 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "qdisp/LargeResultMgr.h"
+
+// System headers
+
+// Third-party headers
+#include "XrdSsi/XrdSsiRequest.hh"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.qdisp.LargeResultMgr");
+
+}
+
+namespace lsst {
+namespace qserv {
+namespace qdisp {
+
+
+/// Set the value of the xrootd semaphore for tracking large response blocks to start.
+void LargeResultMgr::_setup() {
+    std::lock_guard<std::mutex> lck(_mtx);
+    for (int j = 0; j < _runningCountMax; ++j) {
+        _post("initialization");
+    }
+}
+
+
+/// Check if any large result blocks can be run.
```

Comment:
I use block count when debugging and I think it has future value in that it would be a useful measure to provide to an external scheduler/monitor to indicate how busy this system is with large results.